test/distributed/org/apache/cassandra/distributed/test/ReadRepairQueryTester.java

```diff
@@ -0,0 +1,280 @@
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.service.reads.repair.ReadRepairStrategy;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertEquals;
+import static org.apache.cassandra.distributed.shared.AssertUtils.assertRows;
+import static org.apache.cassandra.service.reads.repair.ReadRepairStrategy.NONE;
+
+/**
+ * Tests basic read repair functionality with different types of query types and schemas.
+ * <p>
+ * Each test verifies that its tested query triggers read repair propagating mismatched rows/columns and row/column
+ * deletions. They also verify that the selected rows and columns are propagated through read repair on missmatch,
+ * and that unselected rows/columns are not repaired.
+ * <p>
+ * The tests are parameterized for:
+ * <ul>
+ *     <li><Both {@code NONE} and {@code BLOCKING} read repair stratregies/li>
+ *     <li>Data to be repaired residing on the query coordinator or a replica</li>
+ *     <li>Data to be repaired residing on memtables or flushed to sstables</li>
+ * </ul>
+ * <p>
+ * All the included tests have a similar behaviour:
+ * <ul>
+ *     <li>Create a keyspace with RF=2 and a table</li>
+ *     <li>Insert some data in only one of the nodes</li>
+ *     <li>Run the tested read query selecting a subset of the inserted columns with CL=ALL</li>
+ *     <li>Verify that the previous read has triggered read repair propagating only the queried columns</li>
+ *     <li>Run the tested read query again but this time selecting all the columns</li>
+ *     <li>Verify that the previous read has triggered read repair propagating the rest of the queried rows</li>
+ *     <li>Delete one of the involved columns in just one node</li>
+ *     <li>Run the tested read query again but this time selecting a column different to the deleted one</li>
+ *     <li>Verify that the previous read hasn't propagated the column deletion</li>
+ *     <li>Run the tested read query again selecting all the columns</li>
+ *     <li>Verify that the previous read has triggered read repair propagating the column deletion</li>
+ *     <li>Delete one of the involved rows in just one node</li>
+ *     <li>Run the tested read query again selecting all the columns</li>
+ *     <li>Verify that the previous read has triggered read repair propagating the row deletions</li>
+ *     <li>Verify the final status of each node and drop the table</li>
+ * </ul>
+ */
+@RunWith(Parameterized.class)
+public abstract class ReadRepairQueryTester extends TestBaseImpl
```

Comment:
naming nit: `ReadRepairQueryTester` -> `ReadRepairQueryTest`

Reply:
`ReadRepairQueryTest` produces [an initialization error](https://app.circleci.com/pipelines/github/adelapena/cassandra/1482/workflows/877e7dd4-6e46-4610-aa3b-5a67cc7b2a9d/jobs/15338), I'm keeping `ReadRepairQueryTester` which is in line with other abstract test classes without tests such as `CQLTester`.